### PR TITLE
Add flags to --profile to enable cpu, memory, and trace profiling

### PR DESCRIPTION
**Pull request by marcinja**

### CODE REVIEW

File: profile/profile.go [hunk 1]
```diff
@@ -148,20 +148,22 @@ func startContinuousLog(dir string, restart func()) {
 	}()
 }
 
-// StartContinuousProfiling will continuously print statistics about the cpu
-// usage, memory usage, and runtime stats of the program.
-func StartContinuousProfile(profileDir string) {
+// StartContinuousProfile will continuously print statistics about the cpu
+// usage, memory usage, and runtime stats of the program, and run an execution
+//logger. Select one (recommended) or more functionalities by passing the
+//corresponding flag(s)
```
Comment (starius) [inline profile/profile.go]: Missing spaces after `//`

File: siad/daemon.go [hunk 2]
```diff
@@ -77,16 +77,32 @@ func processModules(modules string) (string, error) {
 	return modules, nil
 }
 
+// processProfiles checks that the flags given for profiling are valid
```
Comment (starius) [inline siad/daemon.go]: Missing `.` in the end of the comment.

File: siad/daemon.go [hunk 3]
```diff
@@ -77,16 +77,32 @@ func processModules(modules string) (string, error) {
...
+func processProfile(profile string) (string, error) {
+	profile = strings.ToLower(profile)
+	validProfiles := "cmt"
+
+	invalidProfiles := profile
+	for _, p := range validProfiles {
+		invalidProfiles = strings.Replace(invalidProfiles, string(p), "", 1)
+	}
```
Comment (starius) [inline siad/daemon.go:88]: I think, regular expression would work better here:
```golang
if !regexp.MustCompile("[cmt]*").Match(profile) {
  ...
}
```

**Review by starius — COMMENTED** [review]
Looks good to me apart from small comments.

File: siad/daemon.go (hunk 3 above)
Comment (marcinja) [reply siad/daemon.go:88]: That regular expression doesn't catch duplicate flags though. The code here is the same as was used in processModules in the same file also.

Comment (starius) [reply siad/daemon.go:88]: Ack

**Comment by starius** [comment]
Thanks!

**Pull request merged**